examples/server_client.rs

```diff
@@ -3,18 +3,20 @@
 //! Note that in practice you don't want to implement a chat client using UDP.
 use std::io::stdin;
 
-use laminar::{config::NetworkConfig, error::Result, net::UdpSocket, Packet};
+use laminar::{Config, NetworkError, Packet, Socket, SocketEvent};
+use std::thread;
 
-const SERVER: &str = "localhost:12351";
+const SERVER: &str = "127.0.0.1:12351";
 
-fn server() -> Result<()> {
-    let mut socket = UdpSocket::bind(SERVER, NetworkConfig::default())?;
+fn server() -> Result<(), NetworkError> {
+    let (mut socket, packet_sender, event_receiver) = Socket::bind(SERVER, Config::default())?;
+    let _thread = thread::spawn(move || socket.start_polling());
 
     println!("Listening for connections to {}", SERVER);
 
     loop {
-        match socket.recv()? {
-            Some(packet) => {
+        match event_receiver.recv().expect("Should get a message") {
```

Comment:
This will kill the application if any error is encountered. Is that what we want?

Reply:
And yes, I know this is in the examples, but I think our examples should probably be as close to production as possible.

Reply:
That's a good question. I defer to the group on this one.